README.md

```diff
@@ -78,14 +79,21 @@ This comparison is made on a best-effort basis. Please create a PR to fix outdat
 cargo install lychee
 ```
 
-Set an environment variable with your token like so `GITHUB_TOKEN=xxxx`.
+Optional (to avoid being rate limited for GitHub links): set an environment variable with your token
+like so `GITHUB_TOKEN=xxxx`, or use the `--github-token` CLI option. This can also be set in the
+config file.
 
 Run it inside a repository with a `README.md` or specify a file with
 
 ```
 lychee <yourfile>
 ```
 
+### CLI return codes
+
+- `0` for success (all links checked successfully or excluded/skipped as configured)
+- `1` for any failures
```

Comment:
This seems to be true for other errors bubbled up to `main()` (such as invalid value for `--method`)

Reply:
Good point. We could use a different exit codes for pure link errors and config errors.
Proposing `1` for link errors and `2` for config errors.

Reply:
Because of [how it currently works in Rust (return 1 for any Result::Err bubbled up to main)](https://github.com/rust-lang/rust/blob/a85e94927622665a9e9022de0d33a890a2e32d43/library/std/src/process.rs#L1844) and how the lychee code is structured, I propose we reverse that.

E.g. `1` for config errors and unexpected runtime errors, and `2` for link errors. Rationale: it's way simpler to bubble all "unexpected" errors to main using the `Result` enum and rely on Rust to exit the process with status code `1`. Where we can control the status code easily without much effort, e.g. link errors, we can return `2`.

This could be expanded to be `3` for specific config errors in the future, if need be, but it would require extra effort to classify errors and to normalize the `?` handling in `main()/run()`.

I will make the change to show what I mean; we can then discuss if it's OK with you.